common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -405,19 +469,35 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
+                    result = DeadlineRunResult.TIMED_OUT;
```

Comment:
@Scottmitch the diff makes it a little difficult to read but this is inside `runAllTasksFrom()` method which does _not_ call `afterRunningAllTasks`. It is called from `SingleThreadEventloop` to execute `tailTasks` and from `SingleThreadEventExecutor` to execute regular tasks.

>>> The executionTimeCallback is used by SingleThreadedEventLoop to determine if there is any execution time remaining, and this TIMED_OUT condition is only checked periodically anyways so it isn't reliable.

You are correct it is checked periodically. I am just honoring the contract of `runAllTasks` that it should _roughly_ take `<= deadlineNanos` time to execute tasks. If we don't, then for high number of `tailTasks`, we may not be honoring the IO-CPU ratio.
So, I think the logic here makes sense.